recipes/tl-expected/all/conanfile.py

```diff
@@ -37,6 +42,8 @@ def package_id(self):
         self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "tl-expected"
-        self.cpp_info.names["cmake_find_package_multi"] = "tl-expected"
+        self.cpp_info.filenames["cmake_find_package"] = "tl-expected"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "tl-expected"
+        self.cpp_info.names["cmake_find_package"] = "tl"
+        self.cpp_info.names["cmake_find_package_multi"] = "tl"
         self.cpp_info.components["expected"].name = "expected"
```

Comment:
I'm wondering now if it's not dangerous to use too generic names for components. What would happen if a recipe would need at least `pkg_config` generator, and depends on both `tl-expected` and an hypothical `expected` recipe whose official pkgconfig name is `expected.pc`. What would happen?

Reply:
One of them will be overridden (UB). 

We shouldn't _create_ components that aren't already provided by the original libraries... and if there is a collision, the collision was already there, Conan is not creating it, the recipe authors should take an action (suggest a new name for the CMake or PkgConfig files/libraries)

Reply:
But components can be created because library has cmake components (and most of the time components are used in recipes to mimic cmake config files), and it might have side effects on pkg_config generator, that's the point. https://github.com/conan-io/conan/issues/7560

Reply:
Understood, thanks for opening the issue 👍